keras/src/backend/tensorflow/numpy.py

```diff
@@ -2430,3 +2430,29 @@ def correlate(x1, x2, mode="valid"):
 
 def select(condlist, choicelist, default=0):
     return tf.experimental.numpy.select(condlist, choicelist, default=default)
+
+
+def argpartition(x, kth, axis=-1):
+    x = convert_to_tensor(x)
+
+    if standardize_dtype(x.dtype) not in ["int32", "int64"]:
+        x = tf.cast(x, tf.int32)
+
+    x = tf.experimental.numpy.swapaxes(x, axis, -1)
+    bottom_ind = tf.math.top_k(-x, kth + 1)[1]
+
+    def set_to_zero(args):
+        a, i = args
+        updates = tf.reshape(tf.zeros_like(i, dtype=tf.float32), [-1])
+        return tf.tensor_scatter_nd_update(a, i, updates)
+
+    for _ in range(x.ndim - 1):
+        set_to_zero = tf.vectorized_map(
+            set_to_zero, (tf.ones(tf.shape(x)), bottom_ind)
+        )
+    proxy = set_to_zero((tf.ones(tf.shape(x)), bottom_ind))
```

Comment:
This appears to be reimplementing the corresponding part of the JAX code that uses `vmap`. But how did you come with this exactly? I can't quite follow the logic.